### ISSUE TITLE
nohup restic mount &

### ISSUE THREAD

**Issue by nunoperalta** [restic/restic#2419]
Output of `restic version`
--------------------------

restic 0.9.5 compiled with go1.12.4 on linux/amd64

How did you run restic exactly?
-------------------------------

source /restic/.restic.env; **nohup** restic mount /mnt/restic &> /restic/mount.log **& disown**

What backend/server/service did you use to store the repository?
----------------------------------------------------------------

S3

Expected behavior
-----------------

When closing Putty, the process should stay alive.

Actual behavior
---------------

When Putty is closed, process dies.
Same happens when using "phpseclib" (which uses SSH).

From another shell:
> [root@test restic]# ps a | grep "restic mount"
>  6779 pts/1    S+     0:00 grep --color=auto restic mount

(before closing Putty, the process is indeed running in background)

The mount stays active, and I can do "umount -f /mnt/restic" to stop the mount.
I just don't think this is the expected/recommended behavior, though...

Do you have an idea how to solve the issue?
-------------------------------------------

Is Restic in some way ignoring/reverting the "nohup" ?

Any suggestions?

**Comment by tkjacobsen** [comment]
Does nohup work for other processes?

And what distro are you on?

There were a lot of discussion a while back about systemd killing all user processes when user logs out. Don't know how it was fixed.

**Comment by nunoperalta** [comment]
wow... I've just tested again with a "sleep 20" and then with "restic mount", and "nohup" seems to be working perfectly...
I even tested using the "phpseclib" exec() function, and again, works perfectly.

No idea what was happening the other day I was testing, as I tried many many times, but the process was always dying after the shell closes.

I've done a full server upgrade in the meantime, so I wonder if that fixed it.
Apparently, restic also was "locked" for some reason, and I did a lock-clean, which could (or not) be a factor.

At the moment, I'm happy that this seems to be working well.
